src/NetMQ/Core/Patterns/XPub.cs

```diff
@@ -323,13 +346,8 @@ protected override bool XRecv(ref Msg msg)
             if (m_pending.Count == 0)
                 return false;
 
-            msg.Close();
```

Comment:
Agree. But shouldn't we check if it shared? Ot it is never shared at this point?